lib/Sema/MiscDiagnostics.cpp

```diff
@@ -2395,14 +2407,10 @@ void VarDeclUsageChecker::handleIfConfig(IfConfigStmt *ICS) {
 
 /// Apply the warnings managed by VarDeclUsageChecker to the top level
 /// code declarations that haven't been checked yet.
-void swift::performTopLevelDeclDiagnostics(SourceFile &MainFile) {
-  // The AST doesn't have TLCD in sil or library code, so don't diagnose.
-  if (!MainFile.isScriptMode())
-    return;
-  VarDeclUsageChecker checker(MainFile.getASTContext().Diags);
-  for (auto D : MainFile.Decls) {
-    D->walk(checker);
-  }
+void swift::
+performTopLevelDeclDiagnostics(TypeChecker &TC, TopLevelCodeDecl &TLCD) {
```

Comment:
For whatever reason, we don't really pass AST nodes as references, just pointers. Please use that convention for now.